File: reactivestreams/src/main/kotlin/reactivestreams/r2dbc/service/response/UserResponse.java
package reactivestreams.r2dbc.service.response;

import reactivestreams.r2dbc.domain.User;

public record UserResponse(
        Long id,
        String name,
        Integer age
) {
    public static UserResponse of(User user) { return new UserResponse(user.getId(), user.getName(), user.getAge()); }
}

File: reactivestreams/src/main/kotlin/reactivestreams/coroutine/service/response/UserResponse.kt
package reactivestreams.coroutine.service.response

import reactivestreams.coroutine.domain.User

data class UserResponse(
    var id : Long,
    var name : String,
    var age: Int,
) {

    companion object {
        operator fun invoke(user: User) =
            with(user) {
                UserResponse(id!!, name, age)
            }
    }

}